### PR TITLE
[feat] Add Key Development Milestones & Proposed COD #34

### PR DESCRIPTION
**Pull request by jjstnlee**
### Description
[//]: # "[Required] Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- Added and styled the key development milestones
- Added date of completion if completed or "Pending" if not completed instead of "KDM 1", "KDM 2", etc.
- Added icons.tsx file in assets folder

### Screenshots
[//]: # "Add any screenshots that will help your PR reviewer understand what’s going on in the PR!"
[//]: # "[Required] For frontend tasks: add screenshots of the layout and expected behavior (GIFs if you’re feeling extra fancy)!"
[//]: # "Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 
image

## How to review
[//]: # "/[Required] Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"

- Run pnpm dev to check it is working
- Refer to figma design

## Next steps
[//]: # "[Optional] - What has not yet been added to the PR, doesn’t yet have any functionality — essentially, what needs to be fixed or added."
[//]: # "[Optional] any temporary fixes in this PR that should be cleaned up later."

- Spacing between project modal and search bar kinda wack
- I think theres a cleaner way to import the completed/pending icons? I couldn't find the icons that josh used on figma so i just inserted them as raw svgs but theres probably a better/cleaner way to do it (refer to KDM component index file to see what im talking abt) (FIXED)
- USE REM INSTEAD OF PX!!!! (FIXED)

### Online sources
[//]: # "[Optional] insert links to any useful tutorials or documentation that you used for this PR!"

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @itsliterallymonique

### CODE REVIEW

File: components/ProjectModal/index.tsx [hunk 1]
```diff
@@ -73,6 +85,85 @@ export default function ProjectModal({
...
+                icon = (
+                  <svg
+                    xmlns="http://www.w3.org/2000/svg"
+                    width="7"
+                    height="2"
+                    viewBox="0 0 7 2"
+                    fill="none"
+                  >
+                    <path
+                      d="M5.92308 1.5C5.51523 1.5 5.18461 1.16421 5.18461 0.75C5.18461 0.335786 5.51523 0 5.92308 0C6.33092 0 6.66154 0.335786 6.66154 0.75C6.66154 0.948912 6.58374 1.13968 6.44525 1.28033C6.30676 1.42098 6.11893 1.5 5.92308 1.5ZM3.70769 1.5C3.29985 1.5 2.96923 1.16421 2.96923 0.75C2.96923 0.335786 3.29985 0 3.70769 0C4.11553 0 4.44615 0.335786 4.44615 0.75C4.44615 0.948912 4.36835 1.13968 4.22986 1.28033C4.09137 1.42098 3.90354 1.5 3.70769 1.5ZM1.49231 1.5C1.08447 1.5 0.753845 1.16421 0.753845 0.75C0.753845 0.335786 1.08447 0 1.49231 0C1.90015 0 2.23077 0.335786 2.23077 0.75C2.23077 0.948912 2.15297 1.13968 2.01448 1.28033C1.87599 1.42098 1.68816 1.5 1.49231 1.5Z"
```
Comment (ronniebeggs) [inline components/ProjectModal/index.tsx:146]: Is there any way to store these constants in a separate file? perhaps in an `/assets` folder?

File: components/ProjectModal/index.tsx [hunk 2]
```diff
@@ -73,6 +85,85 @@ export default function ProjectModal({
...
+                  milestoneLabel = 'NYSERDA contracted';
+                  kdmNumber = 4;
+                  break;
+                case 'Execution of an Interconnection Agreement (IA)':
+                  milestoneLabel = 'IA executed';
+                  kdmNumber = 5;
+                  break;
+                case 'Start of operations':
+                  milestoneLabel = 'Operations begun';
+                  kdmNumber = 6;
```
Comment (ronniebeggs) [inline components/ProjectModal/index.tsx:118]: May be useful to store milestone titles and labels in some sort of enum structure in case their underlying value changes in the future.

File: components/ProjectModal/index.tsx [hunk 3]
```diff
@@ -73,6 +85,85 @@ export default function ProjectModal({
...
+                    fill="none"
+                  >
+                    <path
+                      d="M7.66153 7H0.769226V6H7.66153V7ZM3.23076 4.71L1.26153 2.71L1.95569 2.005L3.23076 3.295L6.47507 0L7.16923 0.71L3.23076 4.71Z"
+                      fill="#ABBFF0"
+                    />
+                  </svg>
+                );
+              } else {
+                icon = (
```
Comment (ronniebeggs) [inline components/ProjectModal/index.tsx:137]: For code cleanliness sake, can we move some of this javascript code outside of the inline JSX. There's a lot to comprehend here if you don't have full context on what's going on.

File: components/ProjectModal/styles.ts [hunk 4]
```diff
@@ -79,3 +83,55 @@ export const closeButtonStyles: CSSProperties = {
...
+  gap: 12px;
+`;
+
+export const KeyDevelopmentMilestone = styled.div<{ status: boolean }>`
+  padding-top: 8px;
+  padding-left: 15px;
+  width: 127px;
+  height: 30px;
+  flex-shrink: 0;
+  border-radius: 12px 4px 4px 4px;
```
Comment (ronniebeggs) [inline components/ProjectModal/styles.ts:102]: I can't see the actual output but flagging this asymmetry in case this was an accident.

File: components/ProjectModal/styles.ts [hunk 5]
```diff
@@ -79,3 +83,55 @@ export const closeButtonStyles: CSSProperties = {
...
+      ? '0.5px solid #4974E0'
+      : '0.5px solid rgba(46, 58, 89, 0.20)'};
+  border-left: ${props =>
+    props.status
+      ? '0.5px solid #4974E0'
+      : '0.5px solid rgba(46, 58, 89, 0.20)'};
+  background: ${props =>
+    props.status
+      ? 'linear-gradient(0deg, rgba(73, 116, 224, 0.00) 0%, rgba(73, 116, 224, 0.08) 100%)'
+      : 'none'};
```
Comment (ronniebeggs) [inline components/ProjectModal/styles.ts:114]: Given this complexity, it almost seems worthwhile to create a separate react component where this logic is handled more cleanly in js. Could also split this component down further.

File: types/schema.d.ts [hunk 6]
```diff
@@ -14,7 +14,7 @@ export type Project = {
   assembly_district: number;
   project_image: string | null;
   additional_information: string | null;
-  key_development_milestones: object | null;
+  key_development_milestones: [Milestone] | null;
```
Comment (ronniebeggs) [inline types/schema.d.ts:17]: Aren't arrays of items declared as `Milestone[]`? I may be getting confused with java lol

**Review by ronniebeggs — CHANGES_REQUESTED** [review]
Left some brief code cleanup changes!
